### PR TITLE
Adds scalerot.dm

### PR DESCRIPTION
**Pull request by LanCartwright**
Scalerot affects scaled species only. This means lizards and flies.

Scalerot will rot them until they have no scales left, which will revert them to a fleshy form. Cloning them will not regenerate those scales either.

Other races can be carriers.

### CODE REVIEW

**Comment by Buggy123** [comment]
Does it murder as quickly as the human version did last time we tested it?

File: code/datums/diseases/advance/symptoms/scalerot.dm [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+					M << M.say(pick("Hiss?"))
+					M << "<span class='danger'>[pick("You painfully let out a hiss")]</span>"
+					M.adjustCloneLoss(15)
+				if(5)
+					M << "<span class='danger'>[pick("Your scales rot away and reveal flesh")]</span>"
+					M.adjustCloneLoss(20)
+				if(6)
+					M.adjustCloneLoss(25)
+					M << "<span class='danger'>[pick("You tear at your scales and rip off your scaley skin!")]</span>"
+					if(M.stat == DEAD)
```
Comment (Xhuis) [inline code/datums/diseases/advance/symptoms/scalerot.dm]: != dead?

File: code/datums/diseases/advance/symptoms/scalerot.dm [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		if(M.dna.species.id == "lizard"||M.dna.species.id == "fly")
+			switch(A.stage)
+				if(2)
+					M << "<span class='notice'>[pick("Your scales feels awfully itchy", "Your tail hurts")]</span>"
+				if(3)
+					M << "<span class='notice'>[pick("You claw at your scales", "Your claws feel strange")]</span>"
+					M.adjustCloneLoss(10)
+				if(4)
+					M << M.say(pick("Hiss?"))
+					M << "<span class='danger'>[pick("You painfully let out a hiss")]</span>"
```
Comment (Xhuis) [inline code/datums/diseases/advance/symptoms/scalerot.dm]: pick() with only one option is bad

File: code/datums/diseases/advance/symptoms/scalerot.dm (hunk 1 above)
Comment (LanCartwright) [inline code/datums/diseases/advance/symptoms/scalerot.dm]: yes. i got complaints and a veto from kaze if it instantly killed him

File: code/datums/diseases/advance/symptoms/scalerot.dm (hunk 2 above)
Comment (LanCartwright) [inline code/datums/diseases/advance/symptoms/scalerot.dm]: shit. right. yeah. I'll fix that or add some other words.

**Comment by LanCartwright** [comment]
Sort of. Depends. Probably yeah, most of the time.

File: code/datums/diseases/advance/symptoms/scalerot.dm [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		stage_speed = 3
+		transmittable = 0
+		level = 6
+		severity = 5
+
+/datum/symptom/scalerot/Activate(var/datum/disease/advance/A)
+	..()
+	if(prob(SYMPTOM_ACTIVATION_PROB))
+		var/mob/living/carbon/human/M = A.affected_mob
+		if(M.dna.species.id == "lizard"||M.dna.species.id == "fly")
```
Comment (neersighted) [inline code/datums/diseases/advance/symptoms/scalerot.dm]: space operators please

**Comment by GunHog** [comment]
Seriously, please, use a different reagent for all these new symptoms. Do you remember when Goof added all those trash disabilities to Genetics, which made it take significantly longer to cut through the extra RNG in order to create a desirable SE?

_This is the virology version of that!_

**Comment by LanCartwright** [comment]
@GunHog would you like these to be uranium operated?

File: code/datums/diseases/advance/symptoms/scalerot.dm (hunk 1 above)
Comment (Supermichael777) [inline code/datums/diseases/advance/symptoms/scalerot.dm]: uh this has a dead guy writhing in agony and yanking at his skin. rethink what you are trying to do and how you want to do it.

Comment (LanCartwright) [inline code/datums/diseases/advance/symptoms/scalerot.dm]: I'll run it through a % based system.

**Comment by Niknakflak** [comment]
Flies have scales?  Pretty sure they don't

**Comment by duncathan** [comment]
Since when do flies have scales

**Comment by KazeEspada** [comment]
He added flies because I pointed out that flies were immune to both human and lizard versions, making them the true race of powergamers.

**Comment by duncathan** [comment]
lmao

**Comment by Dorsisdwarf** [comment]
>flies hissing

**Comment by Xhuis** [comment]
can we stop with the meme symptoms now thanks

**Comment by LanCartwright** [comment]
@KorPhaeron please remove the DNM. it's been tested and balanced now.

**Comment by Cheridan** [comment]
If you want better 'lore', you could say that the symptom only effects creatures that are cold-blooded.

File: code/datums/diseases/advance/symptoms/scalerot.dm [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+		resistance = 3
+		stage_speed = 2
+		transmittable = 1
+		level = 6
+		severity = 5
+
+/datum/symptom/scalerot/Activate(var/datum/disease/advance/A)
+	..()
+	if(prob(SYMPTOM_ACTIVATION_PROB))
+		var/mob/living/carbon/human/M = A.affected_mob
```
Comment (AnturK) [inline code/datums/diseases/advance/symptoms/scalerot.dm]: Is the default human only ? If not you probably want to restrict this one from monkeys.

File: code/datums/diseases/advance/symptoms/scalerot.dm [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+					M.adjustCloneLoss(15)
+					M.adjustBruteLoss(20)
+					if(M.dna.species.id == "lizard")
+						M << "<span class='danger'>You tear at your scales and rip off some scales!</span>"
+					if(M.dna.species.id == "fly")
+						M << "<span class='danger'>You loosen some chitin and slough off some chitin!</span>"
+					if(prob(16.6))
+						M.set_species(/datum/species/human)
+						M.update_icons()
+						M.update_hair()
```
Comment (AnturK) [inline code/datums/diseases/advance/symptoms/scalerot.dm]: These two are handled by set_species already, no need for them

Comment (LanCartwright) [inline code/datums/diseases/advance/symptoms/scalerot.dm]: I set these up as effectively safety measures. I'll remove them if they're unneeded beyond reasonable doubt.

**Comment by KazeEspada** [comment]
Just close this

**Comment by LanCartwright** [comment]
I need to port my math over on this. Will come later.

**Comment by KorPhaeron** [comment]
Open it/finish it again when it's ready then the ancient PRs are triggering me

**Pull request closed without merging**